lib/pbench/agent/tool_data_sink.py

```diff
@@ -1143,22 +1160,22 @@ def record_tms(self, tms):
 
         section = "pbench"
         mdlog.add_section(section)
-        mdlog.set(section, "config", self.optional_md["config"])
-        mdlog.set(section, "date", self.optional_md["date"])
-        mdlog.set(section, "name", self.params.benchmark_run_dir.local.name)
+        mdlog.set(section, "config", self.optional_md.get("config", ""))
```

Comment:
This represents a real bug.  I'll file a direct change to `main`.

Reply:
Filed PR #2770.

Reply:
Can `self.optional_md` itself be `None`, or is it always at least an empty dictionary?  (I'll try to figure this out, but I figured I would ask first.)

Reply:
OK, nevermind:  it's set to an empty dictionary if it's otherwise unspecified.